### PR TITLE
Separated the user plugin vs plugin writer docs

### LINKED ISSUE #3260: Make plugin developer docs more clearly labeled
There are two small issues currently:

* The user facing "list of plugins" page is stored in the source tree with the docs that are really for plugin writers
* The plugin writer docs are under a heading called "Plugin API", but in practice this contains way more than just the API so it should be renamed.

### PR DESCRIPTION
**Pull request by bmbouter**
closes #3260

### CODE REVIEW

File: docs/glossary.rst [hunk 1]
```diff
@@ -51,7 +51,7 @@ Glossary
 
     pulpcore
         A python package offering users a :doc:`rest_api` and plugin writers a
-        :ref:`Plugin API`. It is :term:`plugin`-based and manages :term:`Content`.
+        :ref:`plugin_api`. It is :term:`plugin`-based and manages :term:`Content`.
```
Comment (dralley) [inline docs/glossary.rst:54]: Should we say more?  The pulpcore API also helps with managing repository versions and so forth.

**Review by dralley — APPROVED** [review]

File: docs/glossary.rst (hunk 1 above)
Comment (bmbouter) [reply docs/glossary.rst:54]: We should, but in another piece of work. I'm hoping to spend more time with our docs as part of my contribution.

File: docs/index.rst [hunk 2]
```diff
@@ -21,9 +21,9 @@ If you want to look at the considerations and requirements for installing Pulp,
  :ref:`installation`. If you want to evaluate Pulp quickly, try `Pulp in One
  Container <https://pulpproject.org/pulp-in-one-container/>`_
 
-If you're looking for documentation specific to a content type, see :doc:`List of Plugins<plugins/index>`
+If you're looking for documentation specific to a content type, see :doc:`List of Plugins<plugins>`
 
-Anyone interested in writing a plugin should reference the :ref:`Plugin API`.
+Anyone interested in writing a plugin should reference the :ref:`plugin_development`.
```
Comment (ggainey) [inline docs/index.rst:26]:
```suggestion
Anyone interested in writing a plugin should reference :ref:`plugin_development`.
```

**Review by ggainey — CHANGES_REQUESTED** [review]
One minor tweak, otherwise this is great!

**Review by ggainey — APPROVED** [review]
lg2m!

**Comment by mdellweg** [comment]
Please rebase onto the latest CI fixes.

**Pull request merged**